Pre-apply the transform when building the `Path2D` objects for Type3-fonts

Snuffleupagus opened the pull request:
Rather than updating the transform every time that we're painting a Type3-glyph, we can instead just compute the "final" coordinates during building of the `Path2D` objects.

Snuffleupagus committed:
Pre-apply the transform when building the `Path2D` objects for Type3-fonts

Rather than updating the transform every time that we're painting a Type3-glyph, we can instead just compute the "final" coordinates during building of the `Path2D` objects.

Snuffleupagus commented:
/botio test

calixteman commented on src/display/canvas.js:
```diff
@@ -451,14 +460,11 @@ function compileType3Glyph(imgData) {
-    c.beginPath();
-    c.restore();
+  const drawOutline = function (ctx) {
+    ctx.save();
```
I don't think we need to save, beginPath and restore here since we don't change any state.
Previously save/restore was required because we changed the transform.
About beginPath, I don't think it was required because the current path isn't changed by fill(path), I wasn't sure of that an d tried:
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <style>
        canvas {
            border: 1px solid black;
        }
    </style>
</head>
<body>
    <canvas id="myCanvas" width="500" height="400"></canvas>
    <script>
        const canvas = document.getElementById('myCanvas');
        const ctx = canvas.getContext('2d');

        // Example: Draw a rectangle
        ctx.fillStyle = 'blue';
        ctx.moveTo(50, 50);
        ctx.lineTo(150, 50);
        ctx.lineTo(150, 150);
        ctx.lineTo(50, 150);
        ctx.closePath();

        const path = new Path2D();
        path.moveTo(150, 150);
        path.lineTo(250, 150);
        path.lineTo(250, 250);
        path.lineTo(150, 250);
        path.closePath();

        ctx.fillStyle = 'red';
        ctx.fill(path);

        ctx.fillStyle = 'blue';
        ctx.fill();
     </script>
</body>
</html>
```
So I think we can safely remove that stuff.
And that'd lead give us the possibility to just remove this function and replace it by the path which can be drawn on the call site of the function.

Snuffleupagus replied:
Thank you!

> About beginPath, I don't think it was required because the current path isn't changed by fill(path), I wasn't sure of that and I tried:

Right, I believe the `beginPath`-part may just be a left-over from before this code used `Path2D`.

Snuffleupagus committed:
Simplify the `compileType3Glyph` function to just return the `Path2D` objects

Originally this function would "manually" invoke the rendering commands for Type3-glyphs, however that was changed some time ago:
 - Initial `Path2D` support was added in PR 14858, but the old code kept for Node.js compatibility.
 - Since PR 15951 we've been using a `Path2D` polyfill in Node.js environments.

Hence, after the previous commit, we can further simplify this function by *directly* returning/using the `Path2D` object when rendering Type3-glyphs; see also https://github.com/mozilla/pdf.js/pull/19731#discussion_r2018712695
While this won't improve performance significantly, when compared to the introduction of `Path2D`, it definately cannot hurt.

calixteman reviewed (approved):
LGTM. Thank you.

Snuffleupagus commented:
/botio botmakeref

Snuffleupagus commented:
/botio makeref
